### ISSUE TITLE
a11y: Update role of Banner component to "live region" where applicable

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#14441]
**Is your feature request related to a problem? Please describe.**

Update role of `Banner` component to "live region" where applicable, like when we have a form that fails validation and displays a banner at the top. For example, creating deployment with a duplicate name

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
